### ISSUE TITLE
High latency causes S3 meta errors

### ISSUE THREAD

**Issue by mneil** [HeliosInteractive/swag-sync#12]
I added a new file to the directory and this immediately fired off.

```
swag-sync.exe Error: 0 : Unable to query S3 for file existence (Error making request with Error Code NotFound and Http Status Code NotFound. No further error information was returned by the service.): swag3/michaelNeil.png
    ProcessId=1109
    ThreadId=71
    DateTime=2016-04-16T00:19:55.2430450Z
```

**Comment by 3p3r** [comment]
After looking at the docs, seems like this is a valid error message. It can happen if upload fails (socket get shut down or causes like that).

**Comment by mneil** [comment]
Is there any way to increase the timeout on the request that causes this error? I think it's checking to see if the file already exists and that is what is timing out?

Also, what is the reason for checking the existence when modifying a file? Regardless of if it exists I would expect a change to the file to upload to S3.

**Comment by 3p3r** [comment]
The check happens after upload async task finishes:
https://github.com/HeliosInteractive/swag-sync/blob/master/swag-sync/swag-sync/Bucket.cs#L170

I am going to open a new issue for adding a timeout to existence check.

**Comment by 3p3r** [comment]
fixed by 878642ed55f75bd60c719b63c7a6a442757105ab
